contracts/non-fungible-tokens/andromeda-auction/src/contract.rs

```diff
@@ -287,10 +331,13 @@ fn execute_update_auction(
     min_bid: Option<Uint128>,
 ) -> Result<Response, ContractError> {
     let ExecuteContext {
-        deps, info, env, ..
+        mut deps,
+        info,
+        env,
+        ..
     } = ctx;
     nonpayable(&info)?;
-    validate_denom(&deps.querier, coin_denom.clone())?;
+    validate_denom(deps.branch(), env.clone(), coin_denom.clone())?;
```

Comment:
Why do Deps need to be mutable here? Should `deps.as_ref()` not be sufficient? I don't think this method writes to storage at all?

Reply:
`is_permissioned_strict` could consume a use in case the permission's limited.